### DISCUSSION TITLE
New site’s launch content

### DISCUSSION THREAD

**Discussion by thibaudcolas** [wagtail/guide#105] [Ideas]
We’re planning for the site’s launch and could use feedback on the proposed content. Here is the current set of pages we’ve implemented, as well as existing proposals for new content:
| Status        | Level 1                                                                 | Level 2                                                                          | Level 3                                                                                                                            | Notes                                       |
| ------------- | ----------------------------------------------------------------------- | -------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------- |
| Migrated      | [Home](https://wagtail-guide-staging.torchbox.dev/en-latest/) (english) |                                                                                  |                                                                                                                                    |                                             |
| New           |                                                                         | [Tutorials](https://wagtail-guide-staging.torchbox.dev/en-latest/tutorials/)     |                                                                                                                                    |                                             |
| Migrated      |                                                                         |                                                                                  | [Getting started](https://wagtail-guide-staging.torchbox.dev/en-latest/tutorials/getting-started/)                                 | Currently more of an intro to other content |
| New           |                                                                         | [How-to](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/)           |                                                                                                                                    |                                             |
| Migrated      |                                                                         |                                                                                  | [Finding your way around](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/finding-your-way-around/)                    |                                             |
| Migrated      |                                                                         |                                                                                  | [Creating and editing pages](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/creating-and-editing-pages/)              |                                             |
| Proposed      |                                                                         |                                                                                  | Locking pages – addition to _Creating and editing pages_                                                                           |                                             |
| Migrated      |                                                                         |                                                                                  | [Copying and aliasing pages](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/copying-and-aliasing-pages/)              |                                             |
| Migrated      |                                                                         |                                                                                  | [Managing documents](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/managing-documents/)                              |                                             |
| Migrated      |                                                                         |                                                                                  | [Managing images](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/managing-images/)                                    |                                             |
| Migrated      |                                                                         |                                                                                  | [Managing snippets](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/managing-snippets/)                                |                                             |
| Migrated      |                                                                         |                                                                                  | [Managing collections](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/managing-collections/)                          |                                             |
| Migrated      |                                                                         |                                                                                  | [Managing redirects](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/managing-redirects/)                              |                                             |
| Migrated      |                                                                         |                                                                                  | [Managing users and roles](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/managing-users-and-roles/)                  |                                             |
| Migrated      |                                                                         |                                                                                  | [Configuring moderation workflows](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/configuring-moderation-workflows/)  |                                             |
| Migrated      |                                                                         |                                                                                  | [How to promote search results](https://wagtail-guide-staging.torchbox.dev/en-latest/how-to/how-to-promote-search-results/)        |                                             |
| Proposed #94  |                                                                         |                                                                                  | Wagtail forms                                                                                                                      |                                             |
| Proposed #96  |                                                                         |                                                                                  | ModelAdmin                                                                                                                         |                                             |
| Proposed #98  |                                                                         |                                                                                  | Groups                                                                                                                             |                                             |
| Proposed #99  |                                                                         |                                                                                  | Account profile & notifications                                                                                                    |                                             |
| New (empty)   |                                                                         | [Explanation](https://wagtail-guide-staging.torchbox.dev/en-latest/explanation/) |                                                                                                                                    |                                             |
| Proposed      |                                                                         |                                                                                  | Accessibility                                                                                                                      |                                             |
| Proposed      |                                                                         |                                                                                  | Glossary                                                                                                                           |                                             |
| Done          |                                                                         | [Reference](https://wagtail-guide-staging.torchbox.dev/en-latest/reference/)     |                                                                                                                                    |                                             |
| New           |                                                                         |                                                                                  | [Wagtail page status definitions](https://wagtail-guide-staging.torchbox.dev/en-latest/reference/wagtail-page-status-definitions/) |                                             |
| Proposed #95  |                                                                         |                                                                                  | Scheduled publishing                                                                                                               |                                             |
| Proposed #104 |                                                                         |                                                                                  | Permissions                                                                                                                        |                                             |
| Proposed #97  |                                                                         |                                                                                  | Reports                                                                                                                            |                                             |
| Proposed      |                                                                         |                                                                                  | Sites                                                                                                                              |                                             |
| Proposed      |                                                                         |                                                                                  | Site settings                                                                                                                      |                                             |
| Proposed      |                                                                         |                                                                                  | Popular packages’ docs                                                                                                             |                                             |
| Migrated      |                                                                         |                                                                                  | [Browser issues](https://wagtail-guide-staging.torchbox.dev/en-latest/reference/browser-issues/)                                   |                                             |
| New #84      |                                                                         | About                                                                            |                                                                                                                                    | Project history, contributors listing       |
| New #68      |                                                                         | Help improve this guide                                                          |                                                                                                                                    | Contribution guidance                       |
| Done          |                                                                         | Search results                                                                   |                                                                                                                                    |                                             |

All of the links above will take you to the WIP pages on our staging site. We’d love any feedback on the current content, as well as on the proposals (#94, #96, #98, #99, #95, #104, #97).

**Comment by lb-** [comment]
A few items come to mind...

- locking pages (might be covered in page editing)
- sites (reference)
- site settings (reference)
- links to popular packages' user facing docs OR general user facing docs (could be just a general listing in the reference section) good for submissions from others. 

As for the explanation section - not sure what that is for but I think it's good to maybe have two additional pages somewhere. 

Terminology/ glossary - non Wagtail specific but more website terms like Headless and SEO, Meta tags. 

Accessibility and Inclusion (not sure on naming) - a set of links for best practices when writing and managing content for a wide range of audience. Why alt tags matter, why heading levels matter, how to write for different audiences.

These are low priority probably but could start small and be added to over time.

**Reply by thibaudcolas** [reply]
Thank you! All of the above sound good to me, I’ve added them to the table at the top.

**Reply by lb-** [reply]
Some guides from Awesome Wagtail

- https://foundation.mozilla.org/en/docs/how-do-i-wagtail/
- https://portal.cca.edu/help/wagtail-documentation/
- https://sites.caltech.edu/
- https://goodinternet.org/cms-manual/intro/intro-to-cms/

**Comment by thibaudcolas** [comment]
FYI for anyone looking into this – I’ve implemented basic versions of #84 and #68, which are must-haves for launch. All other proposed sections I will likely not work on myself for the time being, will instead add them to the list at https://github.com/wagtail/guide/discussions/116#discussioncomment-3931928.

**Comment by softkeldozy** [comment]
[responding to this ](https://github.com/wagtail/guide/discussions/116#discussioncomment-3948667)

[The development](https://docs.wagtail.org/en/stable/contributing/developing.html) setup found under the `Contributing` section which according to the Diataxis framework is not supposed to be. It, therefore, seems like the best fit to extract it and all of its contents contained within and place it under the `Reference` section in the new user guide. 

In addition, the steps found [here ](https://docs.wagtail.org/en/stable/contributing/developing.html) for setting up Wagtail locally aren't all that clean and straightforward as compared to using this

- create a folder and give it a name of your choice
- open git bash or any CLI of choice
- `cd` into `your-dir-name`
- Inside `your-dir-name` folder, 
- Run
`git clone https://github.com/wagtail/wagtail.git` or fork it from the repo before cloning if you wish to make a pull request with the repo
- Inside `your-dir-name` folder, again
- Run
`git clone https://github.com/wagtail/bakerydemo.git`  or fork it from the repo before cloning if you wish to make a pull request with the repo.
- Successfully cloning both repositories.
 `cd` bakerydemo

- Check the Python version (Wagtail supports Python 3.7, 3.8, 3.9, 3.10, and 3.11.)
`py --version`

- Install and Activate Virtual Environment 
```
pip install virtualenv
py -m venv .venv
source .venv/Scripts/activate
```

- Install Requirements and Testing Docs
```
pip install -r requirements/development.txt
pip install -e '../wagtail[testing,docs]' -U
```

- Migrate Database and Run Server
```
 ./manage.py migrate
 ./manage.py load_initial_data
 ./manage.py runserver
```

In a separate terminal, `cd` into the Wagtail folder that was cloned in earlier steps (you can open it with any editor of choice). Your path should look like this: "...your-dir-name/wagtail"

- Run the commands below to compile all assets
```
npm ci
npm run build
npm run start
```
make a change in your wagtail and reload the bakerydemo to view the changes!

_Though these steps are basically for windows users and using the virtualenv method in setting bakerydemo locally. I'm using them for clarifying my point._

Some other sections too (like the committing code and UI style guide section under the [contributing to Wagtail](https://docs.wagtail.org/en/stable/contributing/developing.html) can also be moved to the Reference section too)

**Reply by lb-** [reply]
Hey @softkeldozy I might be missing something but this looks like instructions to develop on Wagtail itself and not related to this Guide site. 

Maybe you are best to add this to a different discussion where we are looking at how to make it easier for beginners to get started with development on Wagtail. 
https://github.com/wagtail/wagtail/discussions/9557

**Comment by softkeldozy** [comment]
Also as part of the resources for the development section, the following can be added
- Wagtail's API
- Streamfield
- Jinja template support
- Viewsets

**Reply by lb-** [reply]
See above. This discussion is about what should appear on the Guide for the launch content and not for what should be in the main Wagtail docs. Normal Wagtail users would not need to know these concepts.
